app/controllers/pieces_controller.rb

```diff
@@ -50,6 +52,19 @@ def current_player_controls_piece?(piece)
       !piece.is_white? && piece.game.player_two == current_user
   end
 
+  def current_players_turn?(game)
+    last_piece_moved = game.pieces.order('updated_at').last.moves.order('updated_at').last
+    return true if last_piece_moved.nil? && game.player_one == current_user
+    return false if last_piece_moved.nil?
+    return true if game.player_one == current_user && last_piece_moved.start_piece > 5
+    return true if game.player_two == current_user && last_piece_moved&.start_piece < 6
+    return false
+  end
+
+  def game_has_two_players?(game)
+    return !game.player_one.nil? && !game.player_two.nil?
```

Comment:
This might be clearer if we used `present?` instead of `nil?`. 
As in: `game.player_one.present? && game.player_two.present?`